integrationtests/tests/tests.rs

```diff
@@ -320,7 +321,95 @@ async fn drop_peers_who_contribute_bad_sigs() {
 }
 
 #[tokio::test(flavor = "multi_thread")]
-async fn lightning_gateway_pays_invoice() {
+async fn lightning_gateway_pays_internal_invoice() {
+    let (fed, sending_user, bitcoin, gateway, lightning) =
+        fixtures(2, &[sats(10), sats(100), sats(1000)]).await;
+
+    // Fund the gateway so it can route internal payments
+    fed.mine_and_mint(&gateway.user, &*bitcoin, sats(2000))
+        .await;
+    fed.mine_and_mint(&sending_user, &*bitcoin, sats(2000))
+        .await;
+
+    let receiving_user = sending_user.new_client(&[0]);
+
+    let confirmed_invoice = tokio::join!(
+        receiving_user
+            .client
+            .generate_invoice(sats(1000), "".into(), rng()),
+        fed.await_consensus_epochs(1),
+    )
+    .0
+    .unwrap();
+    let incoming_contract_id = confirmed_invoice.contract_id();
+    let invoice = confirmed_invoice.invoice;
+    info!("Receiving User generated invoice: {:?}", invoice);
+
+    let (contract_id, funding_outpoint) = sending_user
+        .client
+        .fund_outgoing_ln_contract(invoice, rng())
+        .await
+        .unwrap();
+    fed.run_consensus_epochs(2).await; // send coins to LN contract
+
+    let contract_account = sending_user
+        .client
+        .ln_client()
+        .get_contract_account(contract_id)
+        .await
+        .unwrap();
+    assert_eq!(contract_account.amount, sats(1010)); // 1% LN fee
+    info!(
+        "Sending User created outgoing contract: {:?}",
+        contract_account
+    );
+
+    sending_user
+        .client
+        .await_outgoing_contract_acceptance(funding_outpoint)
+        .await
+        .unwrap();
+    info!("Outgoing contract accepted");
+
+    tokio::join!(gateway.server.pay_invoice(contract_id), async {
+        // buy preimage from offer, decrypt preimage, claim outgoing contract, mint the tokens
+        fed.await_consensus_epochs(4).await;
+    })
+    .0
+    .unwrap();
+    info!("Gateway paid invoice on behalf of Sending User");
+
+    gateway
+        .server
+        .await_outgoing_contract_claimed(contract_id, funding_outpoint)
+        .await
+        .unwrap();
+    info!("Gateway claimed outgoing contract");
```

Comment:
Probably don't need all these info! debugs anymore

Reply:
yeah, can probably be a debug message

Reply:
making them debugs